lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -123,32 +130,77 @@ public byte[] Call(byte[] memento)
                         }
                     }
                 }
-                catch (OperationCanceledException e)
-                {
-                    Logger.Log(Level.Warning, "Received OperationCanceledException in MapTaskHost with message: {0}.", e.Message);
-                    break;
-                }
-                catch (IOException e)
+            }
+            catch (OperationCanceledException e)
+            {
+                Logger.Log(Level.Warning,
+                    "Received OperationCanceledException in MapTaskHost with message: {0}. The cancellation token is: {1}.",
+                    e.Message,
+                    _cancellationSource.IsCancellationRequested);
+            }
+            catch (Exception e)
+            {
+                if (e is IOException || e is TcpClientConnectionException || e is RemotingException ||
+                    e is SocketException)
                 {
-                    Logger.Log(Level.Error, "Received IOException in MapTaskHost with message: {0}.", e.Message);
+                    Logger.Log(Level.Error,
+                        "Received Exception {0} in MapTaskHost with message: {1}. The cancellation token is: {2}.",
+                        e.GetType(),
+                        e.Message,
+                        _cancellationSource.IsCancellationRequested);
                     if (!_cancellationSource.IsCancellationRequested)
                     {
+                        Logger.Log(Level.Error,
+                            "MapTask is throwing IMRUTaskGroupCommunicationException with cancellation token: {0}.",
+                            _cancellationSource.IsCancellationRequested);
                         throw new IMRUTaskGroupCommunicationException(TaskManager.TaskGroupCommunicationError);
                     }
-                    break;
                 }
-                catch (TcpClientConnectionException e)
+                else if (e is AggregateException)
```

Comment:
You are catching AggregatonException here also and wrapping it up in IMRU exception and then in TaskManager we are again checking got Aggregation exception....This whole exception handling seems very confusing and needs to be made simpler.